pep-0670.rst

```diff
@@ -38,202 +34,132 @@ to maintain.
-supported by the compiler (GCC and LLVM Clang support it): optimize
-debugging experience. Otherwise, the ``-O0`` compiler option is used:
-disable most optimizations.
-
-With GCC 11, ``gcc -Og`` can inline static inline functions, whereas
-``gcc -O0`` does not inline static inline functions. Examples:
-
-* Call ``Py_INCREF()`` in ``PyBool_FromLong()``:
-
-  * ``gcc -Og``: inlined
-  * ``gcc -O0``: not inlined, call ``Py_INCREF()`` function
-
-* Call ``_PyErr_Occurred()`` in ``_Py_CheckFunctionResult()``:
-
-  * ``gcc -Og``: inlined
-  * ``gcc -O0``: not inlined, call ``_PyErr_Occurred()`` function
-
-On Windows, when Python is built in debug mode by Visual Studio, static
-inline functions are not inlined.
-
-
-Force inlining
---------------
-
-The ``Py_ALWAYS_INLINE`` macro can be used to force inlining. This macro
-uses ``__attribute__((always_inline))`` with GCC and Clang, and
-``__forceinline`` with MSC.
-
-Previous attempts to use ``Py_ALWAYS_INLINE`` didn't show any benefit, and were
-abandoned. See for example: `bpo-45094 <https://bugs.python.org/issue45094>`_:
-"Consider using ``__forceinline`` and ``__attribute__((always_inline))`` on
-static inline functions (``Py_INCREF``, ``Py_TYPE``) for debug build".
-
-When the ``Py_INCREF()`` macro was converted to a static inline
-functions in 2018 (`commit
-<https://github.com/python/cpython/commit/2aaf0c12041bcaadd7f2cc5a54450eefd7a6ff12>`__),
-it was decided not to force inlining. The machine code was analyzed with
-multiple C compilers and compiler options: ``Py_INCREF()`` was always
-inlined without having to force inlining. The only case where it was not
-inlined was the debug build. See discussion in the `bpo-35059
-<https://bugs.python.org/issue35059>`_: "Convert ``Py_INCREF()`` and
-``PyObject_INIT()`` to inlined functions".
-
-
-Disable inlining
-----------------
-
-On the other side, the ``Py_NO_INLINE`` macro can be used to disable
-inlining.  It can be used to reduce the stack memory usage, or to prevent
-inlining on LTO+PGO builds, which are generally more aggressive to inline
-code: see `bpo-33720 <https://bugs.python.org/issue33720>`_. The
-``Py_NO_INLINE`` macro uses ``__attribute__ ((noinline))`` with GCC and
-Clang, and ``__declspec(noinline)`` with MSC.
-
-
 Specification
 =============
 
 Convert macros to static inline functions
 -----------------------------------------
 
-Most macros should be converted to static inline functions to prevent
-`macro pitfalls`_.
+Most macros will be converted to static inline functions.
 
-The following macros should not be converted:
+The following macros will not be converted:
+
+* Object-like macros (i.e. those which don't need parentheses and
+  arguments). For example:
+
+  * Empty macros. Example: ``#define Py_HAVE_CONDVAR``.
+  * Macros only defining a value, even if a constant with a well defined
+    type would be better. Example: ``#define METH_VARARGS 0x0001``.
+  * Py_PYTHON_H
```

Comment:
Just add Py_PYTHON_H as another example on the first item "Empty macros. Example: #define Py_HAVE_CONDVAR.", no?

Reply:
```suggestion
```
“used for definitions rather than behavior” is a better reason to exclude this one, IMO. It does fit both :)